gha: docker: pull image as part of the dependencies

This will avoid the following failure
```
Run bash tests/integration/docker/gha-run.sh run
[gha-run.sh:23] INFO: Running docker smoke test tests using qemu hypervisor
[gha-run.sh:27] INFO: Running docker with runc
Unable to find image 'instrumentisto/nmap:latest' locally
latest: Pulling from instrumentisto/nmap
docker: Get "https://registry-1.docker.io/v2/instrumentisto/nmap/manifests/sha256:f7bdee436758a5f60395684b23924545f5d7478463f20fb86a8671265810435f": read tcp 10.10.1.4:60196->34.226.69.105:443: read: connection reset by peer.
See 'docker run --help'.
Error: Process completed with exit code 125.
```

--- a/tests/integration/docker/gha-run.sh
+++ b/tests/integration/docker/gha-run.sh
@@ -12,11 +12,14 @@ set -o pipefail
 kata_tarball_dir="${2:-kata-artifacts}"
 docker_dir="$(dirname "$(readlink -f "$0")")"
 source "${docker_dir}/../../common.bash"
+image="${image:-instrumentisto/nmap:latest}"
 
 function install_dependencies() {
 	info "Installing the dependencies needed for running the docker smoke test"
 
 	install_docker
+
+	sudo -E docker pull "${image}"
 }
 
 function run() {
@@ -25,10 +28,10 @@ function run() {
 	enabling_hypervisor
 
 	info "Running docker with runc"
-	sudo docker run --rm --entrypoint nping instrumentisto/nmap --tcp-connect -c 2 -p 80 www.github.com
+	sudo docker run --rm --entrypoint nping "${image}" --tcp-connect -c 2 -p 80 www.github.com
 
 	info "Running docker with Kata Containers (${KATA_HYPERVISOR})"
-	sudo docker run --rm --runtime io.containerd.kata-${KATA_HYPERVISOR}.v2 --entrypoint nping instrumentisto/nmap --tcp-connect -c 2 -p 80 www.github.com
+	sudo docker run --rm --runtime io.containerd.kata-${KATA_HYPERVISOR}.v2 --entrypoint nping "${image}" --tcp-connect -c 2 -p 80 www.github.com
 }
 
 function main() {

